modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java

```diff
@@ -258,8 +258,13 @@ public void start(NHttpClientConnection conn) throws IOException, HttpException
                 }
             }
         }
-                                                            
-        request = getHttpRequest(conn, path);
+
+        boolean needContentLength = !((request.getProtocolVersion().equals(HttpVersion.HTTP_1_0))
```

Comment:
needContentLength is getting false if http version is 1.0, which is not correct. Looks like we have mistakenly use the variable name as needContentLegth.  Using needContentLength variable defined here in getHttpRequest() method will not work as intended. Looks like we need to carefully re-write the logic.

Reply:
I changed variable name `needContentLength` to `needToProcessChunking`.
This is original meaning of this condition.